lib/matplotlib/tests/test_backend_tk.py

```diff
@@ -20,28 +22,24 @@ def _isolated_tk_test(success_count, func=None):
     in a subprocess. See GH#18261
 
     The decorated function must be fully self-contained, and thus perform
-    all the imports it needs.  Because its source is extracted and run by
-    itself, coverage will consider it as not being run, so it should be marked
-    with ``# pragma: no cover``
+    all the imports it needs.
     """
 
     if func is None:
         return functools.partial(_isolated_tk_test, success_count)
 
-    # Remove decorators.
-    source = re.search(r"(?ms)^def .*", inspect.getsource(func)).group(0)
-
     @functools.wraps(func)
     def test_func():
+        if (sys.platform == "linux" and
+                not _c_internal_utils.display_is_valid()):
+            pytest.skip("$DISPLAY and $WAYLAND_DISPLAY are unset")
+        elif not importlib.util.find_spec('tkinter'):
+            pytest.skip("missing tkinter")
```

Comment:
I suppose this was needed because the backend decorator needed to be removed to run correctly in the subprocess? Could decorating `test_func` with it do the same?